src/mp.c

```diff
@@ -61,6 +63,14 @@ mp_new(void)
     return z;
 }
 
+MPNumber
+mp_new_from_integer(uint64_t x)
+{
+    MPNumber z;
+    mpc_init2(z.num, PRECISION);
+    mpc_set_si(z.num, x, MPC_RNDNN);
```

Comment:
Wrong datatype (sign and size) of 2nd argument passed to mpc_set_si

See page 9 http://www.multiprecision.org/downloads/mpc-1.2.0.pdf

Reply:
thanks, should be fixed now.